### ISSUE TITLE
dotnet test with msbuild parameter broken in .NET 7

### ISSUE THREAD

**Issue by jooooel** [dotnet/sdk#29227]
### Describe the bug
With .NET 6 I was able to run my tests (without parallelism) using `dotnet test -p:BuildInParallel=false dirs.proj`, however, with .NET 7 this fails with `error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file`.

`dotnet test dirs.proj` works fine, but so does `dotnet msbuild -p:BuildInParallel=false dirs.proj` so I'm not really sure where to file this issue.

Tested on Linux and Mac 

### To Reproduce
Here is a very minimal repo with code that reproduces the issue: https://github.com/jooooel/dotnet-bug

### Exceptions (if any)
`error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file`

### Further technical details
`dotnet-info`

```
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.5
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.100/

Host:
  Version:      7.0.0
  Architecture: arm64
  Commit:       d099f075e4

.NET SDKs installed:
  6.0.200 [/usr/local/share/dotnet/sdk]
  6.0.403 [/usr/local/share/dotnet/sdk]
  7.0.100 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

```

**Comment by baronfel** [comment]
Hello! This is a duplicate of another issue that we've already resolved and will release in the 7.0.101 version of the SDK in early December. Because of this, I'm going to close this issue. Thanks for reporting!

**Comment by jooooel** [comment]
Ah, great! 👍
